### PR TITLE
Validation

### PR DESCRIPTION
**Pull request by barks1212**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,7 +13,8 @@
     "react-hook-form": "6.14.2",
     "react-router-dom": "^5.2.0",
     "react-scripts": "4.0.3",
-    "web-vitals": "^1.0.1"
+    "web-vitals": "^1.0.1",
```
Comment (Jezfx) [inline package.json:16]: ?

File: src/pages/SingleForm/validationSchema.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    .of(Yup.string())
+    .compact()
+    .min(1, "Please Select One"),
+  uk_responsible_person_country: Yup.string().required("Required"),
+  import_countries: Yup.array()
+    .min(1, "Please Include One")
+    .of(Yup.object().shape({ value: Yup.string().min(1, "Too short") })),
+});
+
+export default validationSchema;
```
Comment (Jezfx) [inline src/pages/SingleForm/validationSchema.js]: can we export the resolver instead, something like this: 
```js
import * as Yup from "yup";
import { yupResolver } from "@hookform/resolvers/yup";

const singleFormValidationResolver = yupResolver(Yup.object().shape({
  //..
}));

export default singleFormValidationResolver;

```

File: src/components/ControlledTextField/ControlledTextField.js [hunk 3]
```diff
@@ -18,12 +20,12 @@ const ControlledTextField = ({ name, label, defaultValue }) => {
...
         <TextField
           label={label}
+          error={!!error}
           name={name}
           variant="filled"
           value={value || ""}
           onChange={onChange}
-          // error={!!error}
-          // helperText={error ? error.message : null}
+          helperText={error ? error.message : null}
```
Comment (Jezfx) [inline src/components/ControlledTextField/ControlledTextField.js:28]: can you look at the react warning here

File: src/components/ControlledFieldArray/ControlledFieldArray.js [hunk 4]
```diff
@@ -39,19 +47,21 @@ const ControlledFieldArray = ({ name, label }) => {
...
+              }
             />
           )}
         />
       ))}
       <Box>
         <ButtonGroup color="primary" aria-label="outlined primary button group">
           <Button onClick={appendField}>Add</Button>
-          <Button onClick={removeField}>Remove</Button>
+          {fields?.length >= 1 && <Button onClick={removeField}>Remove</Button>}
```
Comment (Jezfx) [inline src/components/ControlledFieldArray/ControlledFieldArray.js]: thanks for adding this in but can we be more declarative, something like 
```js
const showRemoveButton = Boolean(fields?.length)
```

File: src/components/ControlledFieldArray/ControlledFieldArray.js [hunk 5]
```diff
@@ -39,19 +47,21 @@ const ControlledFieldArray = ({ name, label }) => {
               variant="filled"
               value={value || ""}
               onChange={onChange}
-              // error={!!error}
-              // helperText={error ? error.message : null}
+              error={error && !!error[index]}
+              helperText={
+                error && error[index] ? error[index]?.value?.message : null
```
Comment (Jezfx) [inline src/components/ControlledFieldArray/ControlledFieldArray.js]: like this, helperText can only be used for errors and there would be no way to actually use it for helper text. 
Could do something like this so we have an option to show helper text too
```js

const helperText = getError(error) || field?.helperText
//.. 
<Controller helperText={helperText} />
```

**Review by Jezfx — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (barks1212) [reply package.json:16]: no idea, how this happened

File: src/components/ControlledRadioGroup/ControlledRadioGroup.js [hunk 6]
```diff
@@ -5,14 +5,18 @@ import {
...
 const ControlledRadioGroup = ({ group, name, label }) => {
-  const { control } = useFormContext();
+  const { control, errors } = useFormContext();
   const { options } = group;
 
+  const error = errors[name];
+
   return (
-    <>
+    <FormControl component="fieldset" required error={!!error}>
```
Comment (Jezfx) [inline src/components/ControlledRadioGroup/ControlledRadioGroup.js:19]: required?

**Review by Jezfx — CHANGES_REQUESTED** [review]

**Review by Jezfx — APPROVED** [review]

**Pull request merged**